packages/server/test/react.test.tsx

```diff
@@ -1667,3 +1667,71 @@ describe('withTRPC()', () => {
     });
   });
 });
+
+describe('useQuery() v2', () => {
+  test('no input', async () => {
+    const { trpc, client } = factory;
+    function MyComponent() {
+      const allPostsQuery = trpc.useQuery('allPosts');
+      trpc.useQuery(['allPosts']);
+      expectTypeOf(allPostsQuery.data!).toMatchTypeOf<Post[]>();
+
+      return <pre>{JSON.stringify(allPostsQuery.data ?? 'n/a', null, 4)}</pre>;
+    }
+    function App() {
+      const [queryClient] = useState(() => new QueryClient());
+      return (
+        <trpc.Provider {...{ queryClient, client }}>
+          <QueryClientProvider client={queryClient}>
+            <MyComponent />
+          </QueryClientProvider>
+        </trpc.Provider>
+      );
+    }
+
+    const utils = render(<App />);
+    await waitFor(() => {
+      expect(utils.container).toHaveTextContent('first post');
+    });
+  });
+
+  test('with input', async () => {
+    const { trpc, client } = factory;
+    function MyComponent() {
+      // @ts-expect-error
+      trpc.useQuery('paginatedPosts');
+      trpc.useQuery('paginatedPosts', {
+        input: {
+          limit: 1,
+          // we _should_ add a @ts-expect-error here
+          notExists: 'test',
```

Comment:
@mmkal 👋 

You think we can get this to be erroneous? Right now we allow extra args as `TInput` _`extends`_ something. Feel free to edit on this branch!

I'd like to change this to:

```suggestion
          // @ts-expect-error
          notExists: 'test',
```